main.tf

```diff
@@ -143,6 +143,9 @@ locals {
   cluster_http_load_balancing_enabled        = "${local.cluster_type_output_http_load_balancing_enabled[local.cluster_type] ? false : true}"
   cluster_horizontal_pod_autoscaling_enabled = "${local.cluster_type_output_horizontal_pod_autoscaling_enabled[local.cluster_type] ? false : true}"
   cluster_kubernetes_dashboard_enabled       = "${local.cluster_type_output_kubernetes_dashboard_enabled[local.cluster_type] ? false : true}"
+
+  cluster_basic_auth_username = "${var.enable_basic_auth ? var.basic_auth_username : ""}"
+  cluster_basic_auth_password = "${var.enable_basic_auth ? var.basic_auth_password : ""}"
 }
```

Comment:
Based on those two lines, if `enable_basic_auth` is `true` and `basic_auth_username` and `basic_auth_password` are not specified by the user, based on the `google_container_resource` behaviour, Basic Auth will be disabled (counter-intuitive): maybe in that case we can generate default credentials ?

```
locals {
   basic_auth_username = "${var.basic_auth_username == "" ? "admin" : var.basic_auth_username}"
   basic_auth_password = "${var.basic_auth_password == "" ? resource.random_string.password.result : var.basic_auth_password}"
} 

resource "random_string" "password" {
  length = 16
  special = true
}
```

and change the above to:

```
cluster_basic_auth_username = "${var.basic_auth_enabled ? local.basic_auth_username : ""}"
cluster_basic_auth_password = "${var.basic_auth_enabled ? local.basic_auth_password : ""}"
```

Reply:
That is a little confusing. My first pass did autogenerate, but then the password is stored in tf state so I pulled it out, this boolean is a result of that if I recall... but I guess if an end user is setting the password, it'll be in the state anyway.

The other option I saw was no boolean, and use the presence of both fields to enable it. Thats error prone though on first use if someone sets one and doesn't set the other. Their intention would be to enable auth, but it'd still be disabled. I'm pretty sure changing the auth triggers a recreate too, so thats not ideal.

Reply:
I'm in favour of passing through the values. We shouldn't add extra complexity around a feature which we recommend against using.